modules/Movecraft/src/main/java/net/countercraft/movecraft/listener/BlockListener.java

```diff
@@ -122,9 +118,21 @@ public void onPistonEvent(BlockPistonExtendEvent event) {
         Block block = event.getBlock();
         CraftManager.getInstance().getCraftsInWorld(block.getWorld());
         for (Craft tcraft : CraftManager.getInstance().getCraftsInWorld(block.getWorld())) {
-            MovecraftLocation mloc = new MovecraftLocation(block.getX(), block.getY(), block.getZ());
-            if (MathUtils.locIsNearCraftFast(tcraft, mloc) && tcraft.getCruising() && !tcraft.isNotProcessing()) {
-                event.setCancelled(true);
+            if(tcraft == null) continue;
+
+            if (MathUtils.locationInHitBox(tcraft.getHitBox(), block.getLocation())) {
```

Comment:
Why not put if ![blah] continue here?  It'd reduce the complexity and make this easier to read.